Migrate from gulp

berkus opened the pull request.

Commit:
Migrate to yarn

Commit:
Switch from travis to circle

Commit:
Add README

Commit:
Vanquish the gulpfile

Commit:
Update npmignore

InfiniteRain commented on rollup.config.js:
```diff
@@ -0,0 +1,137 @@
+const distDir = 'dist';
+
+// Name of the SDK, will be reflected in the file name of the CDN version.
+const cdnSdkName = 'operation-retrier';
```
We probably don't need CDN versions of the bundle, as they're only ever consumed by our SDK's via NPM.

berkus replied:
kk

berkus replied:
Will keep for the bright future.

InfiniteRain replied:
ok but let's rename the variable to `bundleName` as this lib is not an sdk

InfiniteRain commented on rollup.config.js:
```diff
@@ -0,0 +1,137 @@
+    ],
+    external,
+  },
+  // Browser CDN bundle.
```
If previous comment is valid, then we shouldn't need the CDN bundle configs.

berkus replied:
Will keep for the bright future.

InfiniteRain commented on package.json:
```diff
@@ -19,7 +19,8 @@
     "sourceMapReport:json": "npx source-map-explorer@2.5.2 ./dist/lib.js --json ./dist/lib.report.json",
-    "docs": "npx jsdoc@3.5.5 -c ./jsdoc.json"
+    "docs": "npx jsdoc@3.5.5 -c ./jsdoc.json",
+    "ci": "yarn clean && yarn lint && yarn test:unit && yarn build && yarn sourceMapReport:html && yarn sourceMapReport:json && yarn docs"
```
Didn't we decide to get rid of `yarn test:unit` part?

berkus replied:
Gone.

Commit:
Enable strict TS

Commit:
Add CI command

Commit:
Add more types

Gradual typing, slowly.

Commit:
Extract AsyncRetrier as a separate entity

This fixes serious problems with modern nodejs where
any non-awaited promise rejection causes visible v8
disturbances. AsyncRetrier provides an interface that
allows ONLY awaiting on it.

InfiniteRain commented on rollup.config.js:
```diff
@@ -0,0 +1,137 @@
+    commonjs({
+      // Ignore these two modules as they're optional and the lib that requires
+      // them has fallbacks available.
+      ignore: ['bufferutil', 'utf-8-validate'],
```
this part not needed as twilsock is not a dependency in here

InfiniteRain commented on rollup.config.js:
```diff
@@ -0,0 +1,137 @@
+    commonjs({
+      // Ignore these two modules as they're optional and the lib that requires
+      // them has fallbacks available.
+      ignore: ['bufferutil', 'utf-8-validate'],
```
this part not needed as twilsock is not a dependency in here

InfiniteRain commented on rollup.config.js:
```diff
@@ -0,0 +1,137 @@
+        format: 'cjs',
+        exports: 'named',
+        banner: licenseBanner,
+      },
```
lets add `sourcemap: true` to here.

Commit:
Integrate rollup

Commit:
Add source map explorer report command

Commit:
Add lint command with ESlint and Prettier

Commit:
Add Jest

Commit:
Apply linter fixes

Commit:
Update gitignore

Commit:
Add test commands

Commit:
Apply linter fixes

InfiniteRain reviewed (approved):
lgtm

InfiniteRain commented on rollup.config.js:
```diff
@@ -0,0 +1,130 @@
+// Destination folder.
+const distDir = 'dist';
+
+// Name of the SDK, will be reflected in the file name of the CDN version.
```
change "SDK" to "bundle" in the comment
